src/interp.cc

```diff
@@ -1329,7 +1329,15 @@ Result Thread::Run(int num_instructions) {
       case Opcode::Call: {
         IstreamOffset offset = ReadU32(&pc);
         if (env_->enable_jit) {
-          auto f = jit::compile(this, offset);
+          jit::JITedFunction f = nullptr;
+
+          if (env_->jit_compiled_functions_.find(offset) != env_->jit_compiled_functions_.end()) {
+            f = env_->jit_compiled_functions_[offset];
```

Comment:
You're doing the same lookup twice here. It would probably be better to assign the result of `env_->jit_compiled_functions_.find(offset)` to a local variable, then use that iterator twice to avoid the second lookup.